main/activities.js

```diff
@@ -16,8 +16,8 @@ class Activities {
   push (ctx, activity) {
     this.#activities.push(activity)
     this.#activities.splice(0, this.#activities.length - 100)
-    ctx.recordActivity(activity)
     this.#detectChangeInOnlineStatus(activity)
+    ctx.recordActivity(activity)
```

Comment:
needed to move this after `#detectChangeInOnlineStatus`, so that when the activity is emitted the online status is already correct. Otherwise we'd stay offline until the next activity is recorded. I thought the asynchronicity of IPC would make this irrelevant, but it does change that.

Reply:
This is very subtle! Have you considered capturing this GH comment in a code comment to make this explanation easier to find in the future?